parsl/monitoring/remote.py

```diff
@@ -218,6 +220,15 @@ def accumulate_and_prepare() -> Dict[str, Any]:
         logging.debug("got children")
 
         d["psutil_cpu_count"] = psutil.cpu_count()
+
+        # note that this will be the CPU number of the base process, not anything launched by it
+        d["psutil_cpu_num"] = pm.cpu_num()
+
+        pctxsw = pm.num_ctx_switches()
+
+        d["psutil_process_num_ctx_switches_voluntary"] = pctxsw.voluntary
+        d["psutil_process_num_ctx_switches_involuntary"] = pctxsw.involuntary
```

Comment:
Inching closer and closer to just getting the current reading from `/proc/self/stat` and storing it for later.

Reply:
there's no "just" there: it needs parsing into the structured database by someone, either the end-user or Parsl takes on that parsing burden.